### PR TITLE
Key agreement

### PR DESCRIPTION
**Pull request by adewaleafolabi**
This PR adds the [keyAgreement](https://www.w3.org/TR/did-core/#key-agreement) field to the knox go library.

Using this in the wallet produces 

```json

{
   "@context":[
      "https://www.w3.org/ns/did/v1",
      "https://w3id.org/security/suites/ed25519-2020/v1"
   ],
   "id":"z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ",
   "authentication":[
      {
         "id":"did:knox:z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ#z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ",
         "type":"Ed25519VerificationKey2020",
         "controller":"did:knox:z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ",
         "publicKeyMultibase":"z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ"
      }
   ],
   "capabilityInvocation":[
      {
         "id":"did:knox:z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ#z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ",
         "type":"Ed25519VerificationKey2020",
         "controller":"did:knox:z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ",
         "publicKeyMultibase":"z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ"
      }
   ],
   "capabilityDelegation":[
      {
         "id":"did:knox:z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ#z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ",
         "type":"Ed25519VerificationKey2020",
         "controller":"did:knox:z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ",
         "publicKeyMultibase":"z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ"
      }
   ],
   "assertionMethod":[
      {
         "id":"did:knox:z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ#z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ",
         "type":"Ed25519VerificationKey2020",
         "controller":"did:knox:z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ",
         "publicKeyMultibase":"z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ"
      }
   ],
   "keyAgreement":[
      {
         "id":"did:knox:z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ",
         "type":"X25519KeyAgreementKey2019",
         "controller":"did:knox:z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ",
         "publicKeyMultibase":"z6MkqFcAnLjLZBDym6bRHUuxDqU3NrdDui7cPwr6aM1tpMBQ"
      }
   ]
}

```

### CODE REVIEW

**Comment by fkim7** [comment]
Can we update the desc to something about implementing https://www.w3.org/TR/did-core/#key-agreement?

Also are we expecting did test suite updates and results as well?

File: helpers/did/did.go [hunk 1]
```diff
@@ -43,5 +43,13 @@ func CreateDidDocument(kps *crypto.KeyPairs) *model.DidDocument {
 				PublicKeyMultibase: kps.AssertionMethodPublicKey,
 			},
 		},
+		KeyAgreement: []model.KeyMaterial{
```
Comment (fkim7) [inline helpers/did/did.go:46]: any tests we can add?

**Comment by LuisOsta** [comment]
@adewaleafolabi What do you think about the testing stuff that Francis brought up?

File: helpers/did/did.go (hunk 1 above)
Comment (adewaleafolabi) [reply helpers/did/did.go:46]: sure we can test that the key agreement is being added to the generated did

**Comment by fkim7** [comment]
FYI just realized this PR belongs to ticket https://github.com/knox-networks/gomono/issues/1269 instead - updating

**Review by fkim7 — CHANGES_REQUESTED** [review]
Please attach did test suite results

**Review by LuisOsta — APPROVED** [review]

**Review by fkim7 — APPROVED** [review]

**Pull request merged**